src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1827,6 +1828,31 @@ mgp_error mgp_vertex_iter_properties(mgp_vertex *v, mgp_memory *memory, mgp_prop
 
 void mgp_edges_iterator_destroy(mgp_edges_iterator *it) { DeleteRawMgpObject(it); }
 
+namespace {
+void NextPermittedEdge(mgp_edges_iterator &it, const bool for_in) {
+  if (!it.source_vertex.graph->ctx || !it.source_vertex.graph->ctx->auth_checker) return;
+
+  auto &impl_it = for_in ? it.in_it : it.out_it;
+  const auto &end = for_in ? it.in->end() : it.out->end();
+
+  if (impl_it) {
+    const auto &auth_checker = it.source_vertex.graph->ctx->auth_checker;
+    const auto db_accessor = *it.source_vertex.graph->ctx->db_accessor;
+    const auto &view = it.source_vertex.graph->view;
```

Comment:
```suggestion
    const auto *auth_checker = it.source_vertex.graph->ctx->auth_checker;
    const auto db_accessor = *it.source_vertex.graph->ctx->db_accessor;
    const auto view = it.source_vertex.graph->view;
```
Reasoning:
1. `auth_checker` is a pointer, by writing `const auto &auth_checker = ...` you introduce another indirection (reference to a pointer), which doesn't have any advantage, but has one drawback: one more indirection. Just copy the pointer and use that.
2. `view` fits into 64 bits, so it makes more sense to copy it, rather then hold a 64bit long pointer and carry the burden of indirection.